examples/graph/app.js

```diff
@@ -0,0 +1,393 @@
+    /* eslint-disable react/no-did-update-set-state */
+    // lastDragged state lasts only one frame
+    const {lastDragged} = this.state;
+    if (lastDragged) {
+      this.setState({
+        lastDragged: null
+      });
+    }
+    /* eslint-enable react/no-did-update-set-state */
+  }
+
+  componentWillUnmount() {
+    this._stopAnimation();
+  }
+
+  _animate() {
+    this.forceUpdate();
+    if (typeof window !== 'undefined') {
+      this._animationFrame = window.requestAnimationFrame(this._animate);
+    }
+  }
+
+  _stopAnimation() {
+    window.cancelAnimationFrame(this._animationFrame);
+    this._animationFrame = null;
+  }
+
+  _resize() {
+    this._onChangeViewport({
+      width: window.innerWidth,
+      height: window.innerHeight
+    });
+  }
+
+  _onChangeViewport(viewport) {
+    this.setState({
+      viewport: {...this.state.viewport, ...viewport}
+    });
+  }
+
+  _onHover(el) {
+    if (el) {
+      this.setState({hovered: el});
+    }
+  }
+
+  _onClick(el) {
+    if (el) {
+      this.setState({clicked: el});
+    } else {
+      const {clicked} = this.state;
+      if (clicked) {
+        this.setState({clicked: null});
+      }
+    }
+  }
+
+  _onDragMove(el) {
+    if (el) {
+      const {viewport} = this.state;
+      const {width, height} = viewport;
+      const x = el.x - width / 2;
+      const y = el.y - height / 2;
+      this.setState({
+        clicked: el,
+        dragging: {
+          node: el.object,
+          x,
+          y
+        }
+      });
+    }
+  }
+
+  _onDragEnd(el) {
+    if (el) {
+      this.setState({
+        dragging: null,
+        lastDragged: {
+          node: el.object
+        }
+      });
+    }
+  }
+
+  //
+  // layout (d3-force) accessors
+  //
+  _linkDistance(link, i) {
+    return 20;
+  }
+
+  _linkStrength(link, i) {
+    if (link.sourceCount || link.targetCount) {
+      return 1 / Math.min(link.sourceCount, link.targetCount);
+    }
+    return 0.5;
+  }
+
+  _nBodyStrength(node, i) {
+    if (node.size) {
+      return -Math.pow(node.size, 1.5) * 3;
+    }
+    return -60;
+  }
+
+  //
+  // rendering (deck.gl) accessors
+  //
+  _getNodeColor(node) {
+    return [18, 147, 154, 255];
+  }
+
+  _getNodeSize(node) {
+    return node.size || 8;
+  }
+
+  // map node type to icon in texture atlas
+  _getNodeIcon(node) {
+    switch (node.type) {
+    case 0:
+      return 'burger';
+    case 1:
+      return 'fries';
+    case 2:
+      return 'soda';
+    case 3:
+    case 4:
+      return 'pie';
+    default:
+      return null;
+    }
+  }
+
+  //
+  // rendering
+  //
+  _renderInteractionLayer(viewport, hovered, clicked) {
+    // set flags used below to determine if SVG highlight elements should be rendered.
+    // if truthy, each flag is replaced with the corresponding element to render.
+    const elements = {
+      hovered: hovered && hovered.object,
+      clicked: clicked && clicked.object
+    };
+    const relatedElements = {
```

Comment:
That is how highlights are applied, yes. However, the `relatedObjects` field only contains the nodes at either end of a selected link, or the links attached to a selected node.

I added the `relatedObjects` field as a convenience for consumers of `graph-layout-layer`. My implementation of choosing elements to highlight, however, is split across `graph-layout-layer::getPickingInfo()` (which assigns `relatedObjects` as I just described), and `app.js::_renderInteractionLayer()`, which goes one step further to pick nodes attached to links that appear as `relatedObjects`.

As I explain this, I'm thinking perhaps I should move all that logic into `getPickingInfo()`. Do you agree that it's a common use case to desire access to nodes connected to a selected node?